### PR TITLE
Some cleanup in aztables

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Moved options and response types to their respective files. Helper funcs that were used in one place have been removed, their contents made inline.
Replaced appending to slices with making the correct size and populating them in place.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to module CHANGELOG.md are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

File: sdk/data/aztables/options.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+	// placeholder for future optional parameters
+}
+
+func (g *GetStatisticsOptions) toGenerated() *generated.ServiceClientGetStatisticsOptions {
+	return &generated.ServiceClientGetStatisticsOptions{}
+}
+
+// ListEntitiesOptions contains optional parameters for Table.Query
+type ListEntitiesOptions struct {
+	// OData filter expression.
```
Comment (chlowell) [inline sdk/data/aztables/options.go:82]: I'm surprised these docstrings pass the linter

**Review by chlowell — APPROVED** [review]

File: sdk/data/aztables/client.go [hunk 2]
```diff
@@ -262,47 +190,25 @@ func (t *Client) NewListEntitiesPager(listOptions *ListEntitiesOptions) *runtime
 			if err != nil {
 				return ListEntitiesResponse{}, err
 			}
-			return newListEntitiesPage(resp)
+			marshalledValue := make([][]byte, 0)
```
[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply sdk/data/aztables/client.go]: Good catch, fixed.

File: sdk/data/aztables/options.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/data/aztables/options.go:82]: I've noticed this before. I suspect that our doc checker isn't properly recursing into struct fields. e.g. some fields in the response envelopes have no doc comments. Opened https://github.com/Azure/azure-sdk-for-go/issues/22446 to track this.

**Pull request merged**